apps/ui/src/components/EditorVote.vue

```diff
@@ -0,0 +1,128 @@
+<script setup lang="ts">
+import Draggable from 'vuedraggable';
+import { BASIC_CHOICES } from '@/helpers/constants';
+import { Draft, VoteType } from '@/types';
+
+const proposal = defineModel<Draft>({ required: true });
+
+defineProps<{
+  error?: string;
+  votingTypes: VoteType[];
+}>();
+
+const VOTING_TYPES_INFO = {
+  basic: {
+    label: 'Basic voting',
+    description: 'Single choice voting with three choices: For, Against or Abstain.'
+  },
+  'single-choice': {
+    label: 'Single choice voting',
+    description: 'Each voter may select only one choice.'
+  },
+  approval: {
+    label: 'Approval voting',
+    description: 'Each voter may select any number of choices.'
+  }
+};
+
+function handleVoteTypeSelected(type: VoteType) {
+  if (!proposal.value) return;
+
+  proposal.value.type = type;
+
+  if (proposal.value.type === 'basic') {
+    proposal.value.choices = [...BASIC_CHOICES];
+  }
+}
+</script>
+
+<template>
+  <template v-if="votingTypes.length > 1 || votingTypes[0] !== 'basic'">
+    <div class="s-base mb-5">
+      <h4 class="eyebrow mb-2.5">Voting type</h4>
+      <div class="flex flex-col gap-[12px]">
+        <button
```

Comment:
Should create a reusable component `UiSelector` and button type should always be `type="button"` because it defaults to `submit` otherwise